src/Microsoft.AspNetCore.Razor/Editor/ImplicitExpressionEditHandler.cs

```diff
@@ -149,10 +154,65 @@ private static bool IsSecondaryDotlessCommitInsertion(Span target, TextChange ch
                    change.OldLength == 0;
         }
 
-        private static bool IsAcceptableReplace(Span target, TextChange change)
+        private bool IsAcceptableReplace(Span target, TextChange change)
         {
             return IsEndReplace(target, change) ||
-                   (change.IsReplace && RemainingIsWhitespace(target, change));
+                (change.IsReplace && RemainingIsWhitespace(target, change));
+        }
+
+        private bool IsAcceptableIdentifierReplacement(Span target, TextChange change)
+        {
+            if (!change.IsReplace)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < target.Symbols.Count; i++)
+            {
+                var symbol = target.Symbols[i] as CSharpSymbol;
+
+                if (symbol == null)
+                {
+                    break;
+                }
+
+                var symbolStartIndex = target.Start.AbsoluteIndex + symbol.Start.AbsoluteIndex;
+                var symbolEndIndex = symbolStartIndex + symbol.Content.Length;
+
+                // We're looking for the first symbol that contains the TextChange.
+                if (symbolEndIndex > change.OldPosition)
+                {
+                    if (symbolEndIndex < change.OldPosition + change.OldLength)
+                    {
+                        // Change spans multiple symbols.
```

Comment:
Can you just invert this condition and add it to the `else if` clause? Then it can just be a simple `if`